storage/kafka/kafka.go

```
@@ -0,0 +1,128 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	kafka "github.com/Shopify/sarama"
+	info "github.com/google/cadvisor/info/v1"
+	storage "github.com/google/cadvisor/storage"
+)
+
+func init() {
+	storage.RegisterStorageDriver("kafka", new)
+}
+
+type kafkaStorage struct {
+	producer    kafka.SyncProducer
+	topic       string
+	machineName string
+}
+
+type detailSpec struct {
+	Timestamp       int64                `json:"timestamp"`
+	MachineName     string               `json:"machine_name,omitempty"`
+	ContainerName   string               `json:"container_Name,omitempty"`
+	ContainerId     string               `json:"container_Id,omitempty"`
+	ContainerLabels map[string]string    `json:"container_labels,omitempty"`
+	ContainerStats  *info.ContainerStats `json:"container_stats,omitempty"`
+}
+
+func (driver *kafkaStorage) containerStatsAndDefaultValues(ref info.ContainerReference, stats *info.ContainerStats) *detailSpec {
+	timestamp := stats.Timestamp.UnixNano() / 1E3
+	var containerName string
+	containerId := ref.Id
+	containerLabels := ref.Labels
+
+	if len(ref.Aliases) > 0 {
+		containerName = ref.Aliases[0]
+	} else {
+		containerName = ref.Name
+	}
+
+	detail := &detailSpec{
+		Timestamp:       timestamp,
+		MachineName:     driver.machineName,
+		ContainerName:   containerName,
+		ContainerId:     containerId,
+		ContainerLabels: containerLabels,
+		ContainerStats:  stats,
+	}
+	return detail
+}
+
+func (driver *kafkaStorage) AddStats(ref info.ContainerReference, stats *info.ContainerStats) error {
+	detail := driver.containerStatsAndDefaultValues(ref, stats)
+	b, _ := json.Marshal(detail)
```

Is there a reason to ignore the error here? I think at the very least you should log it.